### PR TITLE
[docs] feat: restructure support_new_models guide into modular docs with Qwen3-Omni-MoE example

### PR DESCRIPTION
**Pull request by TimYangst**
### What does this PR do?

> Restructures `docs/usage/support_new_models.md` into a modular documentation suite under `docs/usage/support_new_models/`, using Qwen3-Omni-MoE as the primary integration reference alongside the existing Qwen3-VL example.

Key changes:
- Deletes `support_new_models.md` and splits into 3 focused files:
  - `guide_and_checklist.md` — step-by-step integration guide (Steps 0–10) + patch reference + checklists per model category
  - `qwen3_vl_example.md` — Qwen3-VL MoE as a concrete walkthrough
  - `qwen3_omni_moe_example.md` — Qwen3-Omni-MoE as a more complex end-to-end example covering all integration paths
- Updates `docs/index.md` and `README.md` to reference the new structure
- Updates `qwen3_vl_example.md` to align with the actual implementation in `veomni/models/transformers/qwen3_vl_moe/modeling_qwen3_vl_moe.py`: corrects `dummy_forward` (SP-aware grid, `encoder_data_balance` support), deepstack SP (uses `gather_outputs` not `_Gather.apply`), fill-back SP guard (drops `self.training`), fused MoE (correct variable names and `SparseMoeBlock` patch), `get_position_id_func` (adds token ID constants and position ID transposition patch)

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: https://github.com/ByteDance-Seed/VeOmni/pulls?q=docs+support+new+model
- [ ] Format the PR title as `[{modules}] {type}: {description}` ✅

### Test

> Documentation-only change; no code modified. Validated by manual review against the Qwen3-Omni-MoE integration.

### API and Usage Example

N/A — documentation only.

### Design & Code Changes

Files changed:
- `docs/usage/support_new_models.md` — deleted
- `docs/usage/support_new_models/guide_and_checklist.md` — new (+253 lines)
- `docs/usage/support_new_models/qwen3_vl_example.md` — new, updated (+~310 lines)
- `docs/usage/support_new_models/qwen3_omni_moe_example.md` — new (+529 lines)
- `docs/index.md`, `README.md` — updated links

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md?plain=1#L22): `make commit && make style && make quality`
- [ ] Add / Update [the documentation](https://github.com/ByteDance-Seed/VeOmni/blob/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/ByteDance-Seed/VeOmni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: docs-only change, no CI test needed.

### CODE REVIEW

File: docs/usage/support_new_models.md [hunk 1]
```diff
@@ -1,454 +1,971 @@
...
+This single entry exercises:
+- Model loading via registry (`MODELING_REGISTRY`, `MODEL_CONFIG_REGISTRY`)
+- Forward pass under all attention implementations (`eager`, `flash_attention_2`, `flash_attention_3`)
+- Forward pass under all MoE implementations (`eager`, `fused`) if `is_moe=True`
+- Numerical alignment between HF baseline and VeOmni-patched forward
+- Backward pass (gradient flow through all patched components)
 
-# Later, restore these kwargs before calling language_model
-kwargs.update(flash_attn_kwargs)
+If the HF and VeOmni models have different state-dict layouts (e.g. stacked expert weights vs `nn.ModuleList`), add a weight sync function to [tests/models/weight_sync_adapters.py](../../tests/models/weight_sync_adapters.py) and register it under your model's `id`.
```
Comment (Luosuu) [inline docs/usage/support_new_models.md]: let's remove this first as we have not decided how to handle this for huggingface transformers v5

File: docs/usage/support_new_models.md [hunk 2]
```diff
@@ -1,454 +1,971 @@
...
 emb = torch.cat((rotary_pos_emb, rotary_pos_emb), dim=-1)
 position_embeddings = (emb.cos(), emb.sin())
 
-# slice pos embedding if using sp to let sharded hidden_states get its corresponding pos embedding
 if sp_group is not None:
     cos, sin = position_embeddings
-    # Apply same padding and slicing as hidden_states
-    cos = sp_pad_and_slice(cos, dim=0, pad_value=0, pad_scale=4)
-    sin = sp_pad_and_slice(sin, dim=0, pad_value=0, pad_scale=4)
+    cos = sp_pad_and_slice(cos, dim=0, pad_value=0, pad_scale=MERGE_RATIO)
```
Comment (Luosuu) [inline docs/usage/support_new_models.md]: double check

**Comment by Luosuu** [comment]
let's organize like this
```
docs/
|- usage/
|    |- support_new_models/
|    |   |- Guide and Checklist.md
|    |   |- Qwen3-VL-Example.md (original supprt_new_models.md)
|    |   |- Qwen3-Moe-Example.md
```

File: docs/usage/support_new_models.md (hunk 2 above)
Comment (TimYangst) [reply docs/usage/support_new_models.md]: done

File: docs/usage/support_new_models.md (hunk 1 above)
Comment (TimYangst) [reply docs/usage/support_new_models.md]: done

**Comment by TimYangst** [comment]
> let's organize like this
> 
> ```
> docs/
> |- usage/
> |    |- support_new_models/
> |    |   |- Guide and Checklist.md
> |    |   |- Qwen3-VL-Example.md (original supprt_new_models.md)
> |    |   |- Qwen3-Moe-Example.md
> ```

Done

**Review by Coach257 — APPROVED** [review]

File: docs/usage/support_new_models/guide_and_checklist.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
+# Support New Models — Guide and Checklist
```
Comment (piyifan123) [inline docs/usage/support_new_models/guide_and_checklist.md:1]: highlight it's for transformers v4, and v5 needs a slight different patchgen approach for modeling code. Add a TODO for that.

Comment (TimYangst) [reply docs/usage/support_new_models/guide_and_checklist.md:1]: Done
